tests/actionSelector.spec.ts

```diff
@@ -105,4 +107,33 @@ describe("Action selector tests: ", () => {
     expect(actionResponse2).toBeUndefined();
     done();
   });
+
+  it("Should emit refund action", async done => {
+    const actionSelector = new ActionSelector(config);
+    const { entity, actionStub } = setupAction(swapsRefundStub);
+
+    const actionResponse = await actionSelector.selectActions(entity);
+    expect(actionResponse).toStrictEqual(actionStub);
+    done();
+  });
+
+  it("Should emit refund first then redeem action", async done => {
```

Comment:
This is a strange test to me because:
- The issue describes that the refund should be executed only if possible, otherwise execute other action. Here, there is no notion of whether a refund can be executed, yet it is prioritised above the redeem
- This scenario is not possible with Bob
- If this scenario were possible with Bob, then I'd expect the redeem action to be returned (until the refund action can be executed)

Reply:
This scenario is possible if Alice underspends the redeem transaction and it gets canceled but Bob still learns the secret. 
In that case I expect bob to immediately take back his money and only afterwards redeem the other HTLC.